client/src/app/site/pages/organization/pages/committees/pages/committee-list/services/committee-list-sort.service/committee-sort.service.ts

```diff
@@ -23,11 +35,42 @@ export class CommitteeSortService extends BaseSortListService<ViewCommittee> {
     public constructor() {
         super({
             sortProperty: `name`,
-            sortAscending: true
+            sortAscending: true,
+            additionalInfo: {
+                hierarchySort: true
+            }
         });
     }
 
     protected getSortOptions(): OsSortingOption<ViewCommittee>[] {
         return this.staticSortOptions;
     }
+
+    /**
+     * Sorts the given array according to this services sort settings and returns it.
+     */
+    public override async sort(array: ViewCommittee[]): Promise<ViewCommittee[]> {
+        const additionalInfo = (await this.getDefaultDefinition()).additionalInfo;
+        if ((this.additionalInfo as any)?.hierarchySort ?? (additionalInfo as any)?.hierarchySort) {
+            const input = [...array];
+            return (await this.doHierarchySort(input, null)).flat(Infinity);
+        }
+
+        return super.sort(array);
+    }
+
+    private async doHierarchySort(remaining: ViewCommittee[], parentId: Id): Promise<ViewCommittee[]> {
+        const result = [];
+        let i = remaining.length;
+        while (i--) {
```

Comment:
Did you do this instead of a for loop for performance reasons?

Reply:
Yes. As I am removing elements from the `remaining` list within the loop using a while loop is the cleaner way here.